lib/spack/spack/test/module_parsing.py

```diff
@@ -23,21 +23,12 @@
 
 @pytest.fixture
 def module_function_test_mode():
-    old_mode = spack.util.module_cmd._test_mode
-    spack.util.module_cmd._test_mode = True
+    old_cmd = spack.util.module_cmd._cmd_template
+    spack.util.module_cmd._cmd_template = "'. %s 2>&1' % args[1]"
 
     yield
 
-    spack.util.module_cmd._test_mode = old_mode
-
-
-@pytest.fixture
-def save_module_func():
-    old_func = spack.util.module_cmd.module
-
-    yield
-
-    spack.util.module_cmd.module = old_func
+    spack.util.module_cmd._cmd_template = old_cmd
```

Comment:
It seems also the fixtures and test functions in this module could benefit from using `monkeypatch`.